### PR TITLE
Automated observability-pod-disruption-budget tcs

### PR DESCRIPTION
**Pull request by shaior**
Automated observability-pod-disruption-budget (PDB). 
Added replicaset functions to globalhelper (influenced lifecycle suite, fixed in this PR).
Added pod disruption budget to the client API.
Added  clean/ creation/ definition functions for PDB

@greyerof, for TNF - as I have learned, replicasets can be applied as well with PDBs, perhaps we should add it to the test?

### CODE REVIEW

File: tests/globalhelper/replicaset.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		}
+
+		return status
+	}, timeout, retryInterval*time.Second).Should(Equal(true), "replicaSet is not ready")
+
+	return nil
+}
+
+// isReplicaSetReady checks if a replicaset is ready.
+func isReplicaSetReady(namespace string, replicaSetName string) (bool, error) {
```
Comment (sebrandon1) [inline tests/globalhelper/replicaset.go:38]:
```suggestion
func isReplicaSetReady(namespace, replicaSetName string) (bool, error) {
```

File: tests/observability/tests/pod_disruption_budget.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		err = globalhelper.LaunchTests(tnfTestCaseName, qeTcFileName)
+		Expect(err).To(HaveOccurred())
+
+		By("Verify test case status in Junit and Claim reports")
+		err = globalhelper.ValidateIfReportsAreValid(tnfTestCaseName, globalparameters.TestCaseFailed)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	// 56638
+	It("One deployment, pod disruption budget maxUnavailable equals to replica number [negative]", func() {
```
Comment (edcdavid) [inline tests/observability/tests/pod_disruption_budget.go:115]: Should we add a TC where the PDB max unavailable is > maxReplicas

**Review by edcdavid — APPROVED** [review]

File: tests/globalhelper/poddisruptionbudget.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	pdb, err := APIClient.PolicyV1Interface.PodDisruptionBudgets(namespace).Get(context.Background(), pdbName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	return len(pdb.UID) != 0, nil
+}
+
+// CreatPodDisruptionBudget creates Pod Disruption Budget and wait until pdb is created.
+func CreatPodDisruptionBudget(pdb *v1.PodDisruptionBudget, timeout time.Duration) error {
```
Comment (sebrandon1) [inline tests/globalhelper/poddisruptionbudget.go:26]:
```suggestion
func CreatePodDisruptionBudget(pdb *v1.PodDisruptionBudget, timeout time.Duration) error {
```

**Review by sebrandon1 — APPROVED** [review]
Found another typo.

File: tests/observability/tests/pod_disruption_budget.go (hunk 2 above)
Comment (shaior) [reply tests/observability/tests/pod_disruption_budget.go:115]: Added, thanks

**Pull request merged**